### PR TITLE
Upgrade Rubocop to 0.52.1

### LINKED ISSUE #6625: Update dependency constraint to allow for rubocop v0.52.0
Hey there! :wave:

I noticed that the constraint you have for rubocop doesn't allow for the latest version to be used.

The constraint I found was `~> 0.51.0`, and the latest version available is `0.52.0`.

https://github.com/bbatsov/rubocop/releases/tag/v0.52.0

Can you look into updating that constraint so our users can use the latest and greatest version? Thanks! :revolving_hearts:

### PR DESCRIPTION
**Pull request by pathawks**
Fixes #6625

### CODE REVIEW

**Comment by pathawks** [comment]
There are still several cops failing. This will take some work.

Open to any other contributors that have time to work on this before I get back to it :+1:

**Comment by ashmaroli** [comment]
Plenty of false-positives raised with the release of `0.52`.. Might as well wait till they release a `0.52.1` before trying to change our styles..

File: lib/jekyll/configuration.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ class Configuration < Hash
       # Where things are
       "source"              => Dir.pwd,
       "destination"         => File.join(Dir.pwd, "_site"),
-      "collections_dir"     => "",
+     "collections_dir"     => "",
```
Comment (pathawks) [inline lib/jekyll/configuration.rb]: Oh gross, what happened here?

File: lib/jekyll/url.rb [hunk 2]
```diff
@@ -1,17 +1,17 @@
 # frozen_string_literal: true
 
-require "addressable/uri"
+equire "addressable/uri"
```
Comment (pathawks) [inline lib/jekyll/url.rb]: lolwat

File: lib/jekyll/url.rb [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
 # Examples
 #
 #   URL.new({
-#     :template => /:categories/:title.html",
+     :template => /:categories/:title.html",
 #     :placeholders => {:categories => "ruby", :title => "something"}
 #   }).to_s
 #
-module Jekyll
+odule Jekyll
```
Comment (crispgm) [inline lib/jekyll/url.rb]: Also missing a letter here

**Pull request closed without merging**